game-core/src/main/java/games/strategy/debug/console/window/ConsoleModel.java

```diff
@@ -0,0 +1,109 @@
+package games.strategy.debug.console.window;
+
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.util.SystemClipboard;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.java.Log;
+
+@Log
+class ConsoleModel {
+
+  private final ConsoleView consoleView;
+
+  ConsoleModel(final ConsoleView consoleView) {
+    this.consoleView = consoleView;
+
+    // Show console window automatically when the user setting is toggled on.
+    ClientSetting.showConsole.addListener(gameSetting -> {
+      if (gameSetting.getValueOrThrow()) {
+        consoleView.setVisible(true);
+      }
+    });
+
+    // if the console is closed by user manually, keep it closed and do not show it again on startup.
+    consoleView.addWindowClosedListener(() -> ClientSetting.showConsole.setValueAndFlush(false));
+  }
+
+  static String getCurrentLogLevel() {
+    return ClientSetting.loggingVerbosity.getValue()
+        .map(Level::parse)
+        .map(LogLevelItem::fromLevel)
+        .orElse(LogLevelItem.NORMAL.label);
+  }
+
+  static Collection<String> getLogLevelOptions() {
+    return Stream.of(LogLevelItem.values())
+        .map(logLevelSelection -> logLevelSelection.label)
+        .collect(Collectors.toList());
+  }
+
+  void memoryAction() {
+    consoleView.append(DebugUtils.getMemory());
+  }
+
+  void propertiesAction() {
+    consoleView.append(DebugUtils.getProperties());
+  }
+
+  void copyToClipboardAction() {
+    SystemClipboard.setClipboardContents(consoleView.readText());
+  }
+
+  void clearAction() {
+    consoleView.setText("");
+  }
+
+  void enumerateThreadsAction() {
+    consoleView.append(DebugUtils.getThreadDumps());
+  }
+
+  /**
+   * Represents the mapping between user-friendly label and log level.
+   */
+  @VisibleForTesting
+  @AllArgsConstructor
+  @ToString
+  @Getter(AccessLevel.PACKAGE)
+  static class LogLevelItem {
+    static final LogLevelItem DEBUG = new LogLevelItem("Debug Logging", Level.ALL);
+
+    static final LogLevelItem NORMAL = new LogLevelItem("Normal Logging", Level.INFO);
```

Comment:
> My thinking is the 'info' should be relatively clean/useful

I believe that was the case prior to #3715.  But #3715 explicitly dropped all log levels below `INFO`, so that all existing logs below `INFO` were converted to `INFO` as part of that PR.  Now, simply starting a local game with `Normal Logging` enabled dumps **950** lines to the log.

I would suggest either restoring the third `LogLevelItem` or converting the chatty `INFO` logs to `FINE` or lower.  The latter is probably out-of-scope for this PR.

Reply:
https://github.com/triplea-game/triplea/pull/4631 removed `StopWatch` and also the bulk of chatty info logs.